### DISCUSSION TITLE
What brew doctor shall return if brew was successfully installed on OS X for Apple Silicon

### DISCUSSION THREAD

**Discussion by SarvarKh** [Homebrew/discussions#2387] [Getting started] [answered]
I am trying to install **Homebrew** following instructions on setting up open-source repo to my dev environment (OS X for Apple Silicon - Mac M1).
And, their wiki page instructs the following steps.

**1st step** to install homebrew:
`ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"`

**2nd step** run `brew doctor` to check the status of your homebrew install.
 "If successful, you should get: `Your system is ready to brew`"

Following those steps accurately, I am getting the output given below in my terminal after running `brew doctor` (_not the expected result given in the **2nd step**):

```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 13.0.
```

After googling, I understood that the result I am getting is actually the default result to return for homebrew meaning that I have successfully installed homebrew.
Because when I run `brew -v`, I get:
```
Homebrew 3.3.2
Homebrew/homebrew-core (git revision 09010e95556; last commit 2021-10-31)
```
Which means that homebrew is installed properly.

 Please let me know:
(i) if I have installed homebrew successfully (Ideally, please give a reference to homebrew documentation, I couldn't find it myself in brew docs);
(ii) if the expected result `Your system is ready to brew` in the open-source repo is outdated/incorrect for OS X

**Comment by SMillerDev** [comment]
Brew is successfully installed because your brew commands work. But some installs might not work as you expect it until you fix the issues in `brew doctor`.

You seem to have installed the xcode command line tools some time before installing brew, and there is now an update, that is what it's complaining about.

**Reply by SarvarKh** [reply]
Thanks @SMillerDev for prompt help 🙏

**Comment by carlocab** [comment, marked as answer]
You have installed `brew`, but your Command Line Tools are either outdated or missing, which is why `brew doctor` complains. The expected result `Your system is ready to brew.` is correct.

What's the out of `brew config`?

**Reply by SarvarKh** [reply]
`brew config` is returning the following:

```
HOMEBREW_VERSION: 3.3.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9daf5e2d28396c7a4379605e89e029d9fccb41e6
Last commit: 30 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 09010e95556ac305b719fe9c0099e415c1a6b3cf
Core tap last commit: 19 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0.5 build 1205
Git: 2.33.1 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.6-arm64
CLT: 12.5.1.0.1.1623191612
Xcode: N/A
Rosetta 2: false
```

**Reply by carlocab** [reply]
Your CLT is outdated: `CLT: 12.5.1.0.1.1623191612`

You want to install the "Command Line Tools for Xcode 13.0", but you have the one for 12.5.1 installed.

**Reply by SarvarKh** [reply]
@carlocab, **It works!**
I only needed to run `softwareupdate --all --install --force` as `brew doctor` was advising in the terminal.
Not so sure, why it didn't work last time, because I remember I did run this command...
Anyway, thank you for the help 🙏
